custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java

```diff
@@ -182,14 +183,17 @@ public void overrideByPOM(@Nonnull File tmpDir, @Nonnull File pom, final boolean
         }
 
         MavenHelper helper = new MavenHelper(this);
-        String jenkinsVersion = model.getProperties().getProperty("jenkins-war.version");
-        if (StringUtils.isBlank(jenkinsVersion)) {
-            jenkinsVersion = model.getProperties().getProperty("jenkins.version");
-        }
-        if (StringUtils.isNotBlank(jenkinsVersion)) {
-            ComponentReference core = new ComponentReference();
-            core.setVersion(jenkinsVersion);
-            war = core.toWARDependencyInfo();
+        if (bomIncludeWar) {
+            war = null;
+            String jenkinsVersion = model.getProperties().getProperty("jenkins-war.version");
+            if (StringUtils.isBlank(jenkinsVersion)) {
+                jenkinsVersion = model.getProperties().getProperty("jenkins.version");
```

Comment:
I'm not sure it affects, but note that ComponentReference always uses the 'jenkins-war' artifact.

Reply:
It's the artifact used to explode the war. See here: https://github.com/jenkinsci/custom-war-packager/blob/251cc15886f13e92459475cb19ea2281d5f4a28e/custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/config/Config.java#L221